File: rust/axum-postgres/src/telemetry/metrics.rs
use once_cell::sync::Lazy;
use opentelemetry::{
    global,
    metrics::{Counter, Meter},
};

pub static METER: Lazy<Meter> = Lazy::new(|| global::meter("rust-axum-postgres"));

pub static ARTICLES_CREATED: Lazy<Counter<u64>> = Lazy::new(|| {
    METER
        .u64_counter("articles.created")
        .with_description("Total articles created")
        .build()
});

pub static ARTICLES_UPDATED: Lazy<Counter<u64>> = Lazy::new(|| {
    METER
        .u64_counter("articles.updated")
        .with_description("Total articles updated")
        .build()
});

pub static ARTICLES_DELETED: Lazy<Counter<u64>> = Lazy::new(|| {
    METER
        .u64_counter("articles.deleted")
        .with_description("Total articles deleted")
        .build()
});

pub static FAVORITES_ADDED: Lazy<Counter<u64>> = Lazy::new(|| {
    METER
        .u64_counter("favorites.added")
        .with_description("Total favorites added")
        .build()
});

pub static FAVORITES_REMOVED: Lazy<Counter<u64>> = Lazy::new(|| {
    METER
        .u64_counter("favorites.removed")
        .with_description("Total favorites removed")
        .build()
});

pub static USERS_REGISTERED: Lazy<Counter<u64>> = Lazy::new(|| {
    METER
        .u64_counter("users.registered")
        .with_description("Total users registered")
        .build()
});

pub static JOBS_ENQUEUED: Lazy<Counter<u64>> = Lazy::new(|| {
    METER
        .u64_counter("jobs.enqueued")
        .with_description("Total jobs enqueued")
        .build()
});

pub static JOBS_COMPLETED: Lazy<Counter<u64>> = Lazy::new(|| {
    METER
        .u64_counter("jobs.completed")
        .with_description("Total jobs completed successfully")
        .build()
});

pub static JOBS_FAILED: Lazy<Counter<u64>> = Lazy::new(|| {
    METER
        .u64_counter("jobs.failed")
        .with_description("Total jobs failed")
        .build()
});

File: csharp/dotnet-sqlserver/src/Api/Telemetry/Metrics.cs
using System.Diagnostics.Metrics;

namespace Api.Telemetry;

public static class AppMetrics
{
    private static readonly Meter Meter = new("DotnetSqlServer.Metrics");

    public static readonly Counter<long> UsersRegistered =
        Meter.CreateCounter<long>("users.registered", description: "Total users registered");

    public static readonly Counter<long> LoginAttempts =
        Meter.CreateCounter<long>("auth.login.attempts", description: "Total login attempts");

    public static readonly Counter<long> LoginFailures =
        Meter.CreateCounter<long>("auth.login.failures", description: "Total failed login attempts");

    public static readonly Counter<long> ArticlesCreated =
        Meter.CreateCounter<long>("articles.created", description: "Total articles created");

    public static readonly Counter<long> ArticlesUpdated =
        Meter.CreateCounter<long>("articles.updated", description: "Total articles updated");

    public static readonly Counter<long> ArticlesDeleted =
        Meter.CreateCounter<long>("articles.deleted", description: "Total articles deleted");

    public static readonly Counter<long> FavoritesAdded =
        Meter.CreateCounter<long>("favorites.added", description: "Total favorites added");

    public static readonly Counter<long> FavoritesRemoved =
        Meter.CreateCounter<long>("favorites.removed", description: "Total favorites removed");

    public static readonly Counter<long> JobsEnqueued =
        Meter.CreateCounter<long>("jobs.enqueued", description: "Total jobs enqueued");

    public static readonly Counter<long> JobsCompleted =
        Meter.CreateCounter<long>("jobs.completed", description: "Total jobs completed");

    public static readonly Counter<long> JobsFailed =
        Meter.CreateCounter<long>("jobs.failed", description: "Total jobs failed");
}
